simtools/ray_tracing.py

```diff
@@ -108,38 +120,33 @@ def __init__(
 
         self._telescopeModel = self._validateTelescopeModel(telescopeModel)
 
-        self._singleMirrorMode = singleMirrorMode
-        self._useRandomFocalLength = useRandomFocalLength
-
-        # Default parameters
-        if self._singleMirrorMode:
-            collectArguments(
-                self,
-                args=['zenithAngle', 'offAxisAngle'],
-                allInputs=self.ALL_INPUTS,
-                **kwargs
-            )
-            mirFlen = self._telescopeModel.getParameterValue('mirror_focal_length')
-            self._sourceDistance = 2 * float(mirFlen) * u.cm.to(u.km)  # km
-            self._mirrorNumbers = mirrorNumbers
-        else:
-            collectArguments(
-                self,
-                args=['zenithAngle', 'offAxisAngle', 'sourceDistance'],
-                allInputs=self.ALL_INPUTS,
-                **kwargs
-            )
+        # Loading configData
+        _configDataIn = gen.collectDataFromYamlOrDict(configFile, configData)
+        _parameterFile = io.getDataFile('parameters', 'ray-tracing_parameters.yml')
+        _parameters = gen.collectDataFromYamlOrDict(_parameterFile, None)
+        self.config = gen.validateConfigData(_configDataIn, _parameters)
 
         self.label = label if label is not None else self._telescopeModel.label
 
         self._outputDirectory = io.getRayTracingOutputDirectory(self._filesLocation, self.label)
         self._outputDirectory.mkdir(parents=True, exist_ok=True)
 
-        if self._singleMirrorMode:
-            if self._mirrorNumbers == 'all':
+        # Loading relevant attributes in case of single mirror mode.
+        if self.config.singleMirrorMode:
+            # Recalculating source distance.
+            self._logger.debug(
+                'Single mirror mode is activate - source distance is being recalculated to 2 * flen'
+            )
+            mirFlen = self._telescopeModel.getParameterValue('mirror_focal_length')
+            self._sourceDistance = 2 * float(mirFlen) * u.cm.to(u.km)  # km
```

Comment:
Is that also correct for parabolic mirrors? I don't remember, but somehow triggers the thought that the definition of the light source at 2x the focal length is only used for Davies-Cotton optical design.

Reply:
That might be. This feature was written for the MST case.
What would be the source distance for parabolic mirrors then?

Reply:
OK. Not the most efficient way, as the SC MST is much small than the MST, but ok!

Reply:
If you want to simulate an array with SCT instead of MST, you can reduce the sphere when build these layout files. No problem.

But if you want to simulate a mix array with MST and SCT, then you have to set the sphere to the larger radius.

Reply:
> That might be. This feature was written for the MST case.
> What would be the source distance for parabolic mirrors then?

I don't know...always worked with Davies Cotton telescopes. Isn't there an optics paper from MAGIC around? Ask someone there?

Reply:
I will create an issue for that.